packages/block-library/src/table/deprecated.js

```diff
@@ -26,404 +24,643 @@ const oldColors = {
 	'subtle-pale-pink': '#fcf0ef',
 };
 
-const deprecated = [
-	// Deprecation migrating table block to use colors block support feature.
-	{
-		attributes: {
-			hasFixedLayout: {
-				type: 'boolean',
-				default: false,
-			},
-			backgroundColor: {
-				type: 'string',
-			},
-			caption: {
-				type: 'string',
-				source: 'html',
-				selector: 'figcaption',
-				default: '',
-			},
-			head: {
-				type: 'array',
-				default: [],
-				source: 'query',
-				selector: 'thead tr',
-				query: {
-					cells: {
-						type: 'array',
-						default: [],
-						source: 'query',
-						selector: 'td,th',
-						query: {
-							content: {
-								type: 'string',
-								source: 'html',
-							},
-							tag: {
-								type: 'string',
-								default: 'td',
-								source: 'tag',
-							},
-							scope: {
-								type: 'string',
-								source: 'attribute',
-								attribute: 'scope',
-							},
-							align: {
-								type: 'string',
-								source: 'attribute',
-								attribute: 'data-align',
-							},
+const v3 = {
```

Comment:
Thanks for restructuring this into a name version format - it makes it much easier to work with the deprecations this way. It would be good to add a comment above the v3 to outline the reason for the deprecation:

```
// In #41140 support was added to global styles for caption elements which added a `wp-element-caption` classname to the embed figcaption element.
```
this hasn't been common practice in the past, but it helps when looking at deprecations at a glance to work out what they cover.

Reply:
Thank you for your review, done!